.github/workflows/python-package-create.yml

```diff
@@ -1,98 +1,80 @@
-# Creates a Python package via `tox -e package` and uploads it as an artifact named 'packages-<repo name>'.
+# Creates a Python package for an arbitrary repository and subdirectory with optional attestation.
 #######
 
 on:
   workflow_call:
     inputs:
-      python-version:
-        description: "Python version to use; defaults to latest Python release."
-        default: "3.X"
-        type: string
       repository:
         description: "GitHub repository to checkout; defaults to repo running this workflow."
         default: ${{ github.repository }}
         type: string
-      tox-source:
-        description: "The arguments for `pip install` to install tox; use ./path/to/package[dev] for the repo's pinned version."
-        default: ".[dev]"
-        type: string
-      tox-factors:
-        description: "The tox factors to append to the package command."
-        default: ""
-        type: string
       build-subdirectory:
         description: "The subdirectory to build as a wheel."
         default: ""
         type: string
-      distribution-path:
-        description: "Relative filepath to distribution(s); e.g. */dist/*"
-        default: "dist"
-        type: string
-      runner-os:
-        description: "Runner OS to use to run tox; defaults to ubuntu-latest"
-        default: "ubuntu-latest"
+      attest:
+        description: "Create GitHub provenance attestation for the package."
+        default: "false"
         type: string
     outputs:
       artifact-name:
-        description: >
-          Name of the uploaded artifact; use for artifact retrieval.
-          Note that if a `build-subdirectory` is specified, this value will be the "base" of the artifact name.
-          For instance, if the `core` subdirectory of Toga is being built, then this value will be `packages-toga`
-          but the name of the uploaded artifact will be `packages-toga-core`.
-          Therefore, when a `build-subdirectory` is used with this workflow, the `pattern` input for the
-          `actions\download-artifact` should be used to specify `${ needs.package.outputs.artifact-name }-*`.
+        description: "Name of the uploaded artifact; use for artifact retrieval."
         value: ${{ jobs.package.outputs.artifact-name }}
+      artifact-basename:
+        description: >
+          The base name for a group of related package artifacts.
+          When several packages are created from a single repo, such as for Toga, a separate artifact
+          will be created for each package, e.g. Packages-toga-core, Packages-toga-android, etc.
+          Therefore, the base name for these artifacts is Packages-toga. This output allows the caller
+          to retrieve all of these related artifacts with one call to actions/download-artifact.
+        value: ${{ jobs.package.outputs.artifact-basename }}
 
 env:
   FORCE_COLOR: "1"
 
 jobs:
   package:
     name: Create Python Package
-    runs-on: ${{ inputs.runner-os }}
+    runs-on: ubuntu-latest
```

Comment:
Does this mean the rubicon-objc package is being built on Ubuntu? I guess raw packaging shouldn't matter... it just seems a bit odd.

Reply:
Hynek's action is only documented to run on Linux. Presently, it fails on the `tree` command if you don't take steps to install it beforehand.